engine/lib/phx/script/ffi/TimeStamp.lua

```diff
@@ -5,29 +5,52 @@ local TimeStamp
 
 do -- C Definitions
     ffi.cdef [[
-    TimeStamp TimeStamp_Get           ();
-    double    TimeStamp_GetDifference (TimeStamp start, TimeStamp end);
-    double    TimeStamp_GetElapsed    (TimeStamp start);
-    double    TimeStamp_GetElapsedMs  (TimeStamp start);
-    TimeStamp TimeStamp_GetFuture     (double seconds);
-    TimeStamp TimeStamp_GetRelative   (TimeStamp start, double seconds);
-    double    TimeStamp_ToDouble      (TimeStamp);
-  ]]
+        void       TimeStamp_Free          (TimeStamp*);
+        TimeStamp* TimeStamp_Now           ();
+        TimeStamp* TimeStamp_GetFuture     (double seconds);
+        double     TimeStamp_GetDifference (TimeStamp const*, TimeStamp const* end);
+        double     TimeStamp_GetElapsed    (TimeStamp const*);
+        double     TimeStamp_GetElapsedMs  (TimeStamp const*);
+        TimeStamp* TimeStamp_GetRelative   (TimeStamp const*, double seconds);
+        double     TimeStamp_ToDouble      (TimeStamp const*);
+        uint64     TimeStamp_ToSeconds     (TimeStamp const*);
+    ]]
 end
 
 do -- Global Symbol Table
     TimeStamp = {
-        Get           = libphx.TimeStamp_Get,
+        Free          = libphx.TimeStamp_Free,
```

Comment:
Does this need to be heap allocated? Could we teach the proc macro to stack allocate a struct instead?

Reply:
We don't know what is the real size of this type since std::time::SystemTime is platform dependent. On linux it's 96 bit, on other it can be different, so I think it's better to keep it opaque.

Reply:
Right, I understand. That sounds like a good reason to leave it opaque